### ISSUE TITLE
Selecting Elastic Peaks in Engineering Diffraction Fitting Tab Hangs GUI

### ISSUE THREAD

**Issue by ConorMFinn** [mantidproject/mantid#36091]
**Describe the bug**
Selecting an elastic peak when fitting causes the GUI to disable itself and hang.

`Error in execution of algorithm Fit: ExperimentInfo::getEFixed - EFixed requested for elastic mode, don't know what to do!`

This isn't extremely likely to happen as it requires selecting a peak that doesn't make sense to use. But it does disable the GUI entirely requires a restart, which is very annoying when you also lose all setup. 

**To Reproduce**
1. Follow [step 7](https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html#test-7) to load some data into the fitting tab. 
1. When selecting a peak, select an elastic one. 
2. Fit > Fit
3. The gui will be disabled and will not return as the thrown error is caught elsewhere and not handled by the Eng Diff GUI.  

**Expected behavior**
The GUI is re-enabled when the error is thrown. 

**Platform/Version (please complete the following information):**
 - OS: All

**Comment by rbauststfc** [comment]
I'm afraid I'm not familiar with this interface and am struggling to use the provided testing instructions. Conor has kindly supplied the focussed TOF files that I think are needed, but I get the attached error when trying to follow the instructions.

![Error](https://github.com/mantidproject/mantid/assets/38210467/787ee380-e0fa-426b-af2a-1a3e38604d10)

I've tried running through most of the manual testing instructions in case it would create what seems to be missing, but I confess that I stopped at a certain point because it seemed a bit excessive to need to run through all of it to test this small change. I'd be grateful for some more specific instructions to test this fix.
